src/effects/VST/VSTEffect.h

```diff
@@ -199,6 +208,141 @@ struct VSTEffectWrapper : public VSTEffectLink, public XMLTagHandler
    void SaveFXP(const wxFileName& fn) const;
    void SaveFXProgram(wxMemoryBuffer& buf, int index) const;
 
+   // VST plugin -> host callback
+   static intptr_t AudioMaster(AEffect *effect,
+                               int32_t opcode,
+                               int32_t index,
+                               intptr_t value,
+                               void * ptr,
+                               float opt);
+
+   // These are needed to move the AudioMaster callback from VSTEffect to this struct
+   //
+   intptr_t mCurrentEffectID {};
+
+   bool mWantsIdle{ false };
+   std::unique_ptr<VSTEffectTimer> mTimer;
+
+   virtual void NeedIdle() = 0;
+   virtual void UpdateDisplay() = 0;
+   virtual void SizeWindow(int w, int h) = 0;
+
+   virtual VstTimeInfo* GetTimeInfo()
```

Comment:
I would prefer to have these virtual functions defined non-inline, with comments inside the class about what they do (which is mostly nothing)

Reply:
I made them inlinenow.